test/filtering.test.ts

```diff
@@ -0,0 +1,105 @@
+import type { Collection, SanitizedGlobalConfig } from 'payload'
+import { describe, expect, it } from 'vitest'
+import { filterCollections, filterGlobals } from '../src/utils/filtering.js'
+
+describe('Filtering utilities', () => {
+  const mockCollections = {
+    posts: {
+      config: { slug: 'posts', auth: false },
+    } as unknown as Collection,
```

Comment:
Please write the config in a way that actually satisfies the `Collection` type instead of casting to `unknown`